rclcpp/test/test_rate.cpp

```diff
@@ -33,24 +33,24 @@ TEST(TestRate, rate_basics) {
   ASSERT_TRUE(r.sleep());
   auto one = std::chrono::system_clock::now();
   auto delta = one - start;
-  ASSERT_TRUE(period < delta);
-  ASSERT_TRUE(period * overrun_ratio > delta);
+  EXPECT_LT(period, delta);
```

Comment:
Perhaps in a follow-up PR we can help make these checks more robust by checking `period < (delta + constant)`

Reply:
I haven't done that modification yet, because my first assumption is that if you slept for `period`, then `delta` should be always greater than that (considering that the start up time of delta was calculated before rate construction, and the finish time was calculated after the sleep call waked up).

I see that this is only failing when the system timer is being using, and not when using the steady timer. It might be the case that my assumption is only valid for a steady timer.

Reply:
> I see that this is only failing when the system timer is being using, and not when using the steady timer. It might be the case that my assumption is only valid for a steady timer.

For a system timer, you can't reliably assume any time has passed on small timescales; corrections may skew the result.

Reply:
See https://github.com/ros2/rclcpp/pull/1146/commits/c1017e14420ced48bb1dce70c2bdc12885a52252.